src/react-masonry-css.js

```diff
@@ -114,7 +114,7 @@ class Masonry extends React.Component {
     const itemsInColumns = new Array(currentColumnCount);
 
     // Force children to be handled as an array
-    const items = [].concat(this.props.children || []);
+    const items = [].concat(this.props.children || []).filter(child => child);
```

Comment:
Out of curiosity, what does the children you are passing look like to generate the _falsy_ items that you then are filtering out?

I also wonder if [`React.Children.toArray`](https://reactjs.org/docs/react-api.html#reactchildrentoarray) would help with this?

Reply:
It creates an empty column. If you have a 3 column layout, and your children are something like `[real element, falsy, real element]`, the middle column is empty. Maybe it's a false assumption, but I was hoping it would attempt to "left justify" the real children. I have not tried `React.Children.toArray`, will give that a go.